### PR TITLE
Improve high CPU usage and blocking IO detection

### PR DESCRIPTION
**Pull request by vxgmichel**

### CODE REVIEW

File: newsfragments/1124.misc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve high CPU usage and blocking IO detection.
```
Comment (touilleMan) [inline newsfragments/1124.misc.rst:1]: https://xkcd.com/1172/

File: parsec/monitoring.py [hunk 2]
```diff
@@ -11,35 +12,77 @@
...
+    if hasattr(coro, "__class__") and coro.__class__.__name__ in (
+        "async_generator_asend",
+        "async_generator_athrow",
+    ):
+        coro, *_ = gc.get_referents(coro)
+        if not inspect.isasyncgen(coro):
+            return
+
+    # Follow a generator
+    if hasattr(coro, "ag_frame") and coro.ag_frame:
```
Comment (touilleMan) [inline parsec/monitoring.py:31]: `if getattr(coro, "ag_frame", None):`

File: parsec/monitoring.py [hunk 3]
```diff
@@ -11,35 +12,77 @@
...
+        yield from traceback.format_stack(coro.ag_frame)
+        yield from format_stack(coro.ag_await)
+        return
+
+    # Follow a coroutine
+    if hasattr(coro, "cr_frame") and coro.cr_frame:
```
Comment (touilleMan) [inline parsec/monitoring.py:37]: `if getattr(coro, "cr_frame", None):`

File: parsec/monitoring.py (hunk 2 above)
Comment (vxgmichel) [reply parsec/monitoring.py:31]: :+1:

File: parsec/monitoring.py (hunk 3 above)
Comment (vxgmichel) [reply parsec/monitoring.py:37]: :+1:

**Pull request merged**